portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,4 +46,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String serializedJSON = gson.toJson(commentsList);
     response.getWriter().println(serializedJSON);
   }
-}
\ No newline at end of file
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String username = request.getParameter("comment-username");
+    String commentText = request.getParameter("comment-input");
+    commentsList.add(new Comment(username, commentText));
+    response.sendRedirect("/");
+  }
+
+  /** Representation type for a comment, complete with a username and body text */
+  private class Comment {
+
+    private String name;
+    private String text;
```

Comment:
In general, any immutable value should be defined with `final`

Reply:
Now using AutoValue for the data class